diff -up Authen-SASL-2.13/lib/Authen/SASL/Perl.pm.orig Authen-SASL-2.13/lib/Authen/SASL/Perl.pm
--- Authen-SASL-2.13/lib/Authen/SASL/Perl.pm.orig	2013-09-23 13:32:38.956216721 +0200
+++ Authen-SASL-2.13/lib/Authen/SASL/Perl.pm	2013-09-23 13:34:31.713644184 +0200
@@ -274,6 +274,9 @@ sub securesocket {
 
     my $fh = $self->{fh};
 
+    # Fix for BZ#965739, RT#85294
+    $len = length($_[1]) if $len > length($_[1]);
+
     # put on wire in peer-sized chunks
     my $bsz = $self->{sndbufsz};
     while ($len > 0) {
